index.js

```diff
@@ -1,30 +1,42 @@
 import fp from 'fastify-plugin'
 import onFinished from 'on-finished'
-import defaults from 'defaults'
 
 import { DEFAULT_OPTIONS, HEADERS } from './lib/constants.js'
 import { Store } from './lib/store.js'
 import { convertToMs, calculateDelay } from './lib/helpers.js'
 
 const slowDownPlugin = async (fastify, settings) => {
-  const options = defaults(settings, DEFAULT_OPTIONS)
+  const options = { ...DEFAULT_OPTIONS, ...settings }
   const store = new Store(convertToMs(options.timeWindow))
 
+  if (!fastify.hasRequestDecorator('slowDown')) {
+    fastify.decorateRequest('slowDown', undefined)
+  }
```

Comment:
can you do it as described in the docs please instead of coming up with your own way of implementing a request decorator?

Reply:
It is implemented as I found in the documentation.

<img width="848" alt="Screenshot 2022-09-22 at 09 42 33" src="https://user-images.githubusercontent.com/81955552/191682436-771e4727-c5cc-48e8-b9f0-a5c5a601edb6.png">